### PR TITLE
Add docker credential flags

### LINKED ISSUE #57: Add registry auth support from flags
Currently, artefactor will use credentials from the docker client configuration e.g. ~/.docker/config.json or from OSX keychain.

To allow artefactor to be better used in CI / CD pipelines, authentication flags / ENV's should be supported.

### PR DESCRIPTION
**Pull request by lewismarshall**
Allows easy use in CI / CD
Fixes #57.

### CODE REVIEW

File: pkg/cmd/artefactor.go [hunk 1]
```diff
@@ -29,6 +30,14 @@ const (
 	// FlagImageVars is used to specify a whitelist of variable names to enable
 	// "export"
 	FlagImageVars = "image-vars"
+	// FlagDockerUserName overrides docker registry configuration
+	FlagDockerUserName = "docker-username"
+	// FlagDockerPassword overrides docker registry configuration
+	FlagDockerPassword = "docker-password"
+	// FlagDockerPasswordHelp is displayed when getting help for the flag
+	FlagDockerPasswordHelp = "overrides docker registry configuration for password"
+	// FlagDockerPasswordHelp is displayed when getting help for the flag
```
Comment (KashifSaadat) [inline pkg/cmd/artefactor.go]: Comment should be `FlagDockerUserNameHelp`

File: pkg/cmd/artefactor.go [hunk 2]
```diff
@@ -29,6 +30,14 @@ const (
 	// FlagImageVars is used to specify a whitelist of variable names to enable
 	// "export"
 	FlagImageVars = "image-vars"
+	// FlagDockerUserName overrides docker registry configuration
+	FlagDockerUserName = "docker-username"
```
Comment (KashifSaadat) [inline pkg/cmd/artefactor.go]: README should be updated for the new flags with a little note on use-cases.

File: pkg/cmd/artefactor.go [hunk 3]
```diff
@@ -98,6 +107,20 @@ func defaultValue(flagName string, defaultValue string) string {
 	return defaultValue
 }
 
+func getCredsFromFlags(c *cobra.Command) *util.Creds {
+	username := c.Flag(FlagDockerUserName).Value.String()
+	password := c.Flag(FlagDockerPassword).Value.String()
+
+	if len(username) > 0 {
```
Comment (KashifSaadat) [inline pkg/cmd/artefactor.go]: Is an empty password a valid case? Should we be checking it here also?

Comment (lewismarshall) [reply pkg/cmd/artefactor.go]: As the auth for a registry is not our concern and can be handled by any solution (nginx / artefactory etc etc), we should allow it I think.

File: pkg/cmd/artefactor.go (hunk 2 above)
Comment (lewismarshall) [reply pkg/cmd/artefactor.go]: will do

File: pkg/cmd/artefactor.go (hunk 1 above)
Comment (lewismarshall) [reply pkg/cmd/artefactor.go]: Indeed - added #70

Comment (lewismarshall) [reply pkg/cmd/artefactor.go]: nice - not sure why linter didn't catch - added #70

File: pkg/cmd/artefactor.go (hunk 2 above)
Comment (lewismarshall) [reply pkg/cmd/artefactor.go]: Pushed up - please check @KashifSaadat

File: README.md [hunk 4]
```diff
@@ -43,14 +44,16 @@ Artefacts can be specified as flags or as environment variables with the followi
 | `--docker-images` | docker-image docker-image | A white-space delimited set of docker images | `mysql alpine` |
 | `--image-vars` | `"MYSQL_IMAGE ANOTHER_IMAGE"` | A white-space delimited set of image variable names | Given:</br>`export MYSQL_IMAGE=mysql:v5.0`</br>`export ALPINE_IMAGE=alpine` </br> Use: </br>`"MYSQL_IMAGE ALPINE_IMAGE"`|
 | `--web-files` | url,filename,sha256[,true/false] | A white-space separated list of CSV's in the following format: </br></br>`url` is where to download from</br></br> `filename` is the name to save locally</br></br> `sha256` is the expected checksum</br></br>The optional last parameter specifies if the file should have executable permissions | `https://bit.ly/2ySXztI,kd,2f7...,true https://bit.ly/abc.iso,my.iso,abc...` |
-
+| `--docker-username` | `usernmame` | A valid docker registry user-name see # | `bob` |
```
Comment (KashifSaadat) [inline README.md]: `see #`?

Comment (lewismarshall) [reply README.md]: oops - missed that - will make a README fix up later new issue :)

**Pull request merged**